pkgs/development/compilers/ghc/8.10.4.nix

```
@@ -91,7 +91,8 @@ let
 
   # ld.gold is disabled for musl libc due to https://sourceware.org/bugzilla/show_bug.cgi?id=23856
   # see #84670 and #49071 for more background.
-  useLdGold = targetPlatform.isLinux && !(targetPlatform.useLLVM or false) && !targetPlatform.isMusl;
+  useLdGold = (targetPlatform.isLinux && targetPlatform.linker == "bfd" && !targetPlatform.isMusl)
+    || targetPlatform.linker == "gold";
```

I think we can drop Linux case because darwin isn't bfd or gold. I would also like to put the gold condition first because it's the "regular" case.
```suggestion
  useLdGold = targetPlatform.linker == "gold" || (targetPlatform.linker == "bfd" && !targetPlatform.isMusl);
```

---

Done!